testsuite/MDAnalysisTests/transformations/test_translate.py

```diff
@@ -34,28 +34,116 @@
 def translate_universes():
     # create the Universe objects for the tests
     reference = make_Universe(trajectory=True)
-    transformed = make_Universe(trajectory=True)
-    transformed.trajectory.ts.dimensions = np.array([372.,373.,374.,90,90,90])
+    transformed = make_Universe(['masses'], trajectory=True)
+    transformed.trajectory.ts.dimensions = np.array([372., 373., 374., 90, 90, 90])
     return reference, transformed
 
-def test_translate():
-    ref_u, trans_u = translate_universes()
+def test_translate_coords(translate_universes):
+    ref_u = translate_universes[0]
+    trans_u = translate_universes[1]
     ref = ref_u.trajectory.ts
-    vector = np.float32([1,2,3])
+    vector = np.float32([1, 2, 3])
     ref.positions += vector
     trans = translate(vector)(trans_u.trajectory.ts)
     assert_array_almost_equal(trans.positions, ref.positions, decimal=6)
-    
 
-def test_center():
+def test_translate_vector(translate_universes):
+    # what happens if the vector argument is smaller than 3?
+    ts = translate_universes[0].trajectory.ts
+    vector = [0, 1]
+    with pytest.raises(ValueError):
+        translate(vector)(ts)
+    # what happens if the vector argument is bigger than 3?
+    vector = [0, 1, 2, 3]
+    with pytest.raises(ValueError):
+        translate(vector)(ts)
+
+def test_center_in_box_args(translate_universes):
+    # this universe as a box size zero
+    ts = translate_universes[0].trajectory.ts
+    ag = translate_universes[0].residues[0].atoms
+    # what happens if something other than an AtomGroup is given?
+    bad_ag = 1
+    with pytest.raises(ValueError): 
+        center_in_box(bad_ag)(ts)
+    # what if the box is in the wrong format?
+    bad_box = [1, 2, 3]
+    with pytest.raises(ValueError): 
+        center_in_box(ag, box=bad_box)(ts)
+    # is pbc passed to the center methods?
+    # if yes it should raise an exception for boxes that are zero in size
+    with pytest.raises(ValueError): 
+        center_in_box(ag, pbc=True)(ts)
+    # what if a wrong center type name is passed?
+    bad_center = "charge"
```

Comment:
We may implement one day a center of charge. Having something obviously wrong such as "invalid" makes the test future proof and clearer (though, not so much clearer as the variable name is already good).